sdks/java/harness/jmh/src/main/java/org/apache/beam/fn/harness/jmh/control/ExecutionStateSamplerBenchmark.java

```diff
@@ -41,6 +41,16 @@
 public class ExecutionStateSamplerBenchmark {
   private static final String PTRANSFORM = "benchmarkPTransform";
 
+  @State(Scope.Thread)
+  public static class RunnersCoreStateTracker {
+    public ExecutionStateTracker tracker;
+
+    @Setup
+    public void setup(RunnersCoreStateSampler sharedState) {
+      tracker = new ExecutionStateTracker(sharedState.sampler);
+    }
+  }
+
   @State(Scope.Benchmark)
   public static class RunnersCoreStateSampler {
     public final ExecutionStateSampler sampler = ExecutionStateSampler.newForTest();
```

Comment:
I think these states need to be moved to RunnersCoreStateTracker like you did for HarnessStateTracker.

We would only ever expect to have one thread activating/deactivating the state. The race condition that you pointed out was between the sampling thread and one (not many) processing thread.

Reply:
ah yeah, makes sense!